src/libexpr/eval-settings.cc

```diff
@@ -70,11 +70,9 @@ Strings EvalSettings::getDefaultNixPath() const
         }
     };
 
-    if (!restrictEval && !pureEval) {
```

Comment:
Could you add to the commit message how this change fixes the issue? I am trying to understand how this relates to nix-path.

Reply:
Yeah it's still WIP, there will be rationale in the log.

This particular piece gets obsolete because allowed prefixes are handled in the source accessor now.